contracts/Aavegotchi/facets/ERC721BuyOrderFacet.sol

```diff
@@ -0,0 +1,156 @@
+        uint256 erc721TokenId,
+        uint256 indexed category,
+        uint256 priceInWei,
+        uint256 time
+    );
+
+    function getERC721BuyOrder(uint256 _buyOrderId) external view returns (ERC721BuyOrder memory buyOrder_) {
+        buyOrder_ = s.erc721BuyOrders[_buyOrderId];
+        require(buyOrder_.timeCreated != 0, "ERC721BuyOrder: ERC721 buyOrder does not exist");
+    }
+
+    function getERC721BuyOrderByTokenId(uint256 _erc721TokenId) external view returns (ERC721BuyOrder memory buyOrder_) {
+        uint256 buyOrderId = s.erc721BuyOrderHead[_erc721TokenId];
+        require(buyOrderId != 0, "ERC721BuyOrder: buyOrder doesn't exist");
+        buyOrder_ = s.erc721BuyOrders[buyOrderId];
+    }
+
+    function placeERC721BuyOrder(address _erc721TokenAddress, uint256 _erc721TokenId, uint256 _priceInWei) external onlyLocked(_erc721TokenId) {
+        require(_priceInWei >= 1e18, "ERC721BuyOrder: price should be 1 GHST or larger");
+
+        address sender = LibMeta.msgSender();
+        uint256 ghstBalance = IERC20(s.ghstContract).balanceOf(sender);
+        require(ghstBalance >= _priceInWei, "ERC721BuyOrder: Not enough GHST!");
+
+        uint256 category = LibAavegotchi.getERC721Category(_erc721TokenAddress, _erc721TokenId);
+        require(category != LibAavegotchi.STATUS_VRF_PENDING, "ERC721BuyOrder: Cannot buy a portal that is pending VRF");
+        require(sender != s.aavegotchis[_erc721TokenId].owner, "ERC721BuyOrder: Owner can't be buyer");
+
+        require(block.timestamp > s.erc721BuyOrderLocked[_erc721TokenId] + 10 minutes, "ERC721BuyOrder: Buy order locked for this Aavegotchi");
+
+        uint256 oldBuyOrderId = s.erc721BuyOrderHead[_erc721TokenId];
+        if (oldBuyOrderId != 0) {
+            ERC721BuyOrder memory erc721BuyOrder = s.erc721BuyOrders[oldBuyOrderId];
+            require((erc721BuyOrder.cancelled == true) || (erc721BuyOrder.priceInWei < _priceInWei), "ERC721BuyOrder: Higher price buy order already exist");
+            require(erc721BuyOrder.timeCreated != 0, "ERC721BuyOrder: ERC721 buyOrder does not exist");
+            require((erc721BuyOrder.cancelled == false) && (erc721BuyOrder.timePurchased == 0), "ERC721BuyOrder: Already processed");
+
+            LibBuyOrder.cancelERC721BuyOrder(oldBuyOrderId);
+        }
+
+        // Transfer GHST
+        LibERC20.transferFrom(s.ghstContract, sender, address(this), _priceInWei);
+
+        // Place new buy order
+        s.nextERC721BuyOrderId++;
+        uint256 buyOrderId = s.nextERC721BuyOrderId;
+
+        s.erc721BuyOrderHead[_erc721TokenId] = buyOrderId;
+        s.erc721BuyOrders[buyOrderId] = ERC721BuyOrder({
+            buyOrderId: buyOrderId,
+            buyer: sender,
+            erc721TokenAddress: _erc721TokenAddress,
+            erc721TokenId: _erc721TokenId,
+            category: category,
+            priceInWei: _priceInWei,
+            timeCreated: block.timestamp,
+            timePurchased: 0,
```

Comment:
If you're setting to default values, you don't need to include the keys in the struct creation